editoast/src/views/v2/timetable/stdcm.rs

```diff
@@ -274,6 +263,79 @@ async fn stdcm(
     Ok(Json(stdcm_response))
 }
 
+/// Build the list of scheduled train requirements, only including requirements
+/// that overlap with the possible simulation times.
+fn build_train_requirements(
+    trains: Vec<TrainSchedule>,
+    simulations: Vec<(SimulationResponse, PathfindingResult)>,
+    departure_time: DateTime<Utc>,
+    latest_simulation_end: DateTime<Utc>,
+) -> HashMap<i64, TrainRequirement> {
+    let mut trains_requirements = HashMap::new();
+    for (train, sim) in trains.iter().zip(simulations) {
+        let (sim, _) = sim;
+        let final_output = match sim {
+            SimulationResponse::Success { final_output, .. } => final_output,
+            _ => continue,
+        };
+        let spacing_requirements: Vec<SpacingRequirement> = final_output
+            .spacing_requirements
+            .into_iter()
+            .filter(|req| {
+                is_resource_in_range(
+                    departure_time,
+                    latest_simulation_end,
+                    train.start_time,
+                    req.begin_time,
+                    req.end_time,
+                )
+            })
+            .collect();
+        let routing_requirements: Vec<RoutingRequirement> = final_output
+            .routing_requirements
+            .into_iter()
+            .filter(|req| {
+                is_resource_in_range(
+                    departure_time,
+                    latest_simulation_end,
+                    train.start_time,
+                    req.begin_time,
+                    req.zones
+                        .iter()
+                        .map(|zone_req| zone_req.end_time)
+                        .max()
+                        .unwrap_or(req.begin_time),
+                )
+            })
+            .collect();
+        if (!spacing_requirements.is_empty()) || (!routing_requirements.is_empty()) {
```

Comment:
Should we add a pre-check that the train start simulation overlaps the `departure_time` and the `latest_simulation_end`. 

By doing so, we can remove this check and avoid filtering all resources occupied by trains.

Reply:
I've added a pre-check for each train.

It's not as straightforward: a scheduled train can start before the earliest time and still affect the new train (if it lasts long enough). I've kept the resource filtering for the same reason, sometimes only part of the resources are in the range.

But I could remove the last check (that there is at least one overlapping requirement)